### PR TITLE
add definition for bootstrap-tagsinput

### PR DESCRIPTION
**Pull request by itsmattius**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/bootstrap-tagsinput/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	tagsinput(methodName: "add" | "remove", values: any, options?: TagsOptions): JQuery;
+	// removeAll : Removes all tags.
+	// focus : Sets focus in the tagsinput.
+	// input : Returns the tagsinput's internal <input />, which is used for adding tags. You could use this to add your own typeahead behaviour for example.
+	// refresh : Refreshes the tags input UI. This might be usefull when you're adding objects as tags. When an object's text changes, you'll have to refresh to update the matching tag's text.
+	// destroy : Removes tagsinput behaviour.
+	tagsinput(methodName: "removeAll" | "focus" | "input" | "refresh" | "destroy"): JQuery;
+}
+
+declare module "bootstrap-tagsinput" {}
```
Comment (sandersn) [inline types/bootstrap-tagsinput/index.d.ts]: why is this needed?

File: types/bootstrap-tagsinput/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	allowDuplicates?: boolean;
+	// When the input container has focus, the class specified by this config option will be applied to the container.
+	focusClass?: string;
+	// Allow creating tags which are not returned by typeahead's source (default: true).
+	// This is only possible when using string as tags. When itemValue option is set, this option will be ignored.
+	freeInput?: boolean;
+	// Object containing typeahead specific options.
+	typeahead?: Typeahead;
+}
+interface JQuery {
```
Comment (sandersn) [inline types/bootstrap-tagsinput/index.d.ts]: JQuery has two type parameters according to the compile error in the travis logs.

**Review by sandersn — CHANGES_REQUESTED** [review]
Fails to compile; see Travis logs for details.

File: types/bootstrap-tagsinput/index.d.ts (hunk 2 above)
Comment (abedi-ir) [inline types/bootstrap-tagsinput/index.d.ts]: Hi @sandersn 
I tries every way for this error and I have no idea what I should do to resolve this error?
I also looked at the other jquery plugins in this repository and they just use `JQuery` interface.
Can help me?

**Comment by paulvanbrenk** [comment]
Closing as this seems abandoned.

**Pull request closed without merging**